replica/database.cc

```diff
@@ -808,6 +809,26 @@ future<> database::parse_system_tables(distributed<service::storage_proxy>& prox
         }
         co_return;
     });
+    co_await do_parse_schema_tables(proxy, db::schema_tables::AGGREGATES, [&] (schema_result_value_type& v) -> future<> {
+        bool partition_found = false;
+        co_await do_parse_schema_tables(proxy, db::schema_tables::SCYLLA_AGGREGATES, [&] (schema_result_value_type& v2) -> future<> {
+            if (v.first == v2.first) {
+                partition_found = true;
+                auto&& user_aggregates = create_aggregates_from_schema_partition(*this, v.second, v2.second);
+                for (auto&& agg : user_aggregates) {
+                    cql3::functions::functions::add_function(agg);
+                }
+            }
+            co_return;
+        });
+        if (!partition_found) {
+            auto&& user_aggregates = create_aggregates_from_schema_partition(*this, v.second, nullptr);
+            for (auto&& agg : user_aggregates) {
+                cql3::functions::functions::add_function(agg);
```

Comment:
Nitpick: Instead of repeating this code twice, can't we do it just once, here, but just make the second parameter to create_aggregates_from_schema_partition something like `partition_found ? v2.second : nullptr`?

Reply:
Now that I think about it, I should probably just use `read_schema_partition_for_keyspace` here - it will remove both the duplication and even the  `partition_found` boolean